File: Player.cpp
// Player.cpp
#define _CRT_SECURE_NO_WARNINGS
#include "Player.h"
#include <iostream>

Player::Player(const std::string& name) : name(name), score(0) {}

std::string Player::getName() const { return name; }
int Player::getScore() const { return score; }
void Player::setName(const std::string& name) { this->name = name; }
void Player::incrementScore() { score++; }
void Player::resetScore() { score = 0; }
void Player::printInfo() const {
    std::cout << "Player: " << name << ", Score: " << score << std::endl;
}

File: Player.java
// player.java
public class Player {
    private String name;
    private int score;
    private static int playerCount = 0;

    public Player() {
        this.name = "";
        this.score = 0;
        playerCount++;
    }

    public Player(String name) {
        this.name = name;
        this.score = 0;
        playerCount++;
    }

    public Player(String name, int score) { // Новый конструктор для установки очков
        this.name = name;
        this.score = score;
        playerCount++;
    }

    public String getName() {
        return name;
    }

    public int getScore() {
        return score;
    }

    public void incrementScore() {
        this.score++; // Использование оператора this
    }

    public void resetScore() {
        this.score = 0; // Использование оператора this
    }

    public void setName(String name) {
        this.name = name; // Использование оператора this
    }

    public static int getPlayerCount() {
        return playerCount;
    }
}
